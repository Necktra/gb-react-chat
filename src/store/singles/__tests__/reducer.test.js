import {
    getSinglesFailure,
    getSinglesRequest
} from '../actions';
import singlesReducer from './../reducer';
import {
    getSinglesSuccess
} from './../actions';

describe("Singles reducer tests", () => {

    test("returns correct state after GET_SINGLES_REQUEST action", () => {
        const singlesStore = singlesReducer([], getSinglesRequest());
        expect(singlesStore).toMatchSnapshot();
    });

    test("returns correct state after GET_SINGLES_FAILURE action", () => {
        const singlesStore = singlesReducer([], getSinglesFailure("err"));
        expect(singlesStore).toMatchSnapshot();
    });

    test("returns correct state after GET_SINGLES_SUCCESS action", () => {
        const singlesStore = singlesReducer([], getSinglesSuccess("test"));
        expect(singlesStore).toMatchSnapshot();
    });

});