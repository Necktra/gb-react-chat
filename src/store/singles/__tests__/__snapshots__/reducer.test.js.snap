// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Singles reducer tests returns correct state after GET_SINGLES_FAILURE action 1`] = `
Object {
  "error": undefined,
  "request": 3,
}
`;

exports[`Singles reducer tests returns correct state after GET_SINGLES_REQUEST action 1`] = `
Object {
  "error": null,
  "request": 1,
}
`;

exports[`Singles reducer tests returns correct state after GET_SINGLES_SUCCESS action 1`] = `
Object {
  "request": 2,
  "singles": undefined,
}
`;
